app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -386,6 +376,80 @@ class WireActivityViewModel @Inject constructor(
         }
     }
 
+    private suspend fun onConversationInviteDeepLink(
+        code: String,
+        key: String,
+        domain: String?
+    ) =
+        when (val currentSession = coreLogic.getGlobalScope().session.currentSession()) {
+            is CurrentSessionResult.Failure.Generic -> null
+            CurrentSessionResult.Failure.SessionNotFound -> null
+            is CurrentSessionResult.Success -> {
+                coreLogic.sessionScope(currentSession.accountInfo.userId) {
+                    when (val result = conversations.checkIConversationInviteCode(code, key, domain)) {
+                        is CheckConversationInviteCodeUseCase.Result.Success -> {
+                            if (result.isSelfMember) {
+                                // TODO; display messsage that user is already a member and ask if they want to navigate to the conversation
+                                openConversation(result.conversationId)
+                            } else {
+                                globalAppState =
+                                    globalAppState.copy(
+                                        conversationJoinedDialog = JoinConversationViaCodeState.Show(
+                                            result.name,
+                                            code,
+                                            key,
+                                            domain
+                                        )
+                                    )
+                            }
+                        }
+
+                        is CheckConversationInviteCodeUseCase.Result.Failure -> globalAppState =
+                            globalAppState.copy(conversationJoinedDialog = JoinConversationViaCodeState.Error(result))
+                    }
+                }
+            }
+        }
+
+    fun joinConversationViaCode(
```

Comment:
Can we also cover error case with test ? when `currentSession` and `joinConversationViaCode` return Failure